From 61337650a0b540cb4be1e0c9b3ef4e060fde8c93 Mon Sep 17 00:00:00 2001
From: Rishawn <70256146+Stealth1226@users.noreply.github.com>
Date: Mon, 10 Jan 2022 09:47:15 +0530
Subject: [PATCH] base: Disable ripple effect on unlock

    From: Nauval Rizky <enuma.alrizky@gmail.com>
    Date: Wed, 19 Jan 2022 08:31:56 +0700
    Subject: SystemUI: Fix method for disabling unlock ripple
     animation

    NotificationShadeWindowController's are not gracefully closed, which
    leads to top parts of the screen is being obscured by invisible shade window.
    And also, "ripple" is actually still exists due to lightRevealScrim are
    being started by onKeyguardFadingAwayChanged, which is not what user expect
    when they want to disable the effect.

    So in order to handle those cases, lets intercept it from the
    AuthRippleController instead.

    Signed-off-by: alanndz <alanndz7@gmail.com>
    Signed-off-by: Joey Huab <joey@evolution-x.org>
    Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>

[ItsVixano]: Hard disable it :D

Change-Id: Icf288401b6695bc1866d379028402fb7d953f11e
---
 .../android/systemui/biometrics/AuthRippleController.kt   | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/packages/SystemUI/src/com/android/systemui/biometrics/AuthRippleController.kt b/packages/SystemUI/src/com/android/systemui/biometrics/AuthRippleController.kt
index d68fcd0a..a0d7100f 100644
--- a/packages/SystemUI/src/com/android/systemui/biometrics/AuthRippleController.kt
+++ b/packages/SystemUI/src/com/android/systemui/biometrics/AuthRippleController.kt
@@ -161,6 +161,8 @@ class AuthRippleController @Inject constructor(
     }
 
     private fun showUnlockedRipple() {
+        return
+
         notificationShadeWindowController.setForcePluginOpen(true, this)
 
         // This code path is not used if the KeyguardTransitionRepository is managing the light
@@ -185,6 +187,12 @@ class AuthRippleController @Inject constructor(
     }
 
     override fun onKeyguardFadingAwayChanged() {
+        // reset and hide the scrim so it doesn't appears on
+        // the next notification shade usage
+        centralSurfaces.lightRevealScrim?.revealAmount = 1f
+        startLightRevealScrimOnKeyguardFadingAway = false
+        return
+
         if (featureFlags.isEnabled(Flags.LIGHT_REVEAL_MIGRATION)) {
             return
         }
-- 
2.40.0

